packages/core/test/event.test.ts

```diff
@@ -420,4 +420,13 @@ describe('@bugsnag/core/event', () => {
         }))
     })
   })
+
+  describe('event.setTraceCorrelation()', () => {
+    it('allows setting the span and trace id', () => {
```

Comment:
There's a couple of scenarios missing tests:

- if `spanId` is `undefined` or `null` it's still added
- if `traceId` is `undefined` or `null` it's not added

Reply:
Should we also update the api for `setTraceCorrelation` to accept `undefined | null` for the spanId?

Reply:
could they _both_ be expected to be null? 🤔

Reply:
> could they both be expected to be null? 🤔

Presumably we shouldn't call `setTraceCorrelation` if there's no trace ID so I think only span ID should be optional in the API

I don't think it makes sense for span ID to be `null` as an expected value (i.e. in the TS type definition) but it's javascript so it can easily end up that way at runtime

so I think the signature should be:

``ts
setTraceCorrelation(traceId: string, spanId?: string): void
```

but in the method we can't assume they have the right types